### PR TITLE
Revised file-upload logic.

### PR DESCRIPTION
**Pull request by bschroeter**
Due to the API changes from ME.org, the file upload logic needed changing.

Summary:
- API endpoint changed.
- API response changed to immediately return a file_id
  - This negates the need to perform a file_status, but does still require waiting some interval for the file to actually move to the object store.
- The entire client and CLI have been updated and CLI modified with testing.
- Multiple files can be passed to the client on the command line with glob expansion

Originally, we thought that we could upload everything in a directory at once, however, the receiving server has a packet limit, which we have had raised to 200mb. This will allow large (but singular) files to be transferred up to this size. Smaller files can feasibly be sent all at once.

### CODE REVIEW

File: meorg_client/client.py [hunk 1]
```diff
@@ -213,45 +214,67 @@ def logout(self):
             self.headers.pop("X-User-Id", None)
             self.headers.pop("X-Auth-Token", None)
 
-    def get_file_status(self, id: str) -> Union[dict, requests.Response]:
-        """Get the file status.
+    def upload_file(
+        self,
+        file_path: Union[str, list],
+        file_obj: Union[io.BufferedReader, list] = None,
+    ) -> Union[dict, requests.Response]:
```
Comment (SeanBryan51) [inline meorg_client/client.py:221]: I think this function is larger than it needs to be. Have you considered using a list of file inputs where a file input is either a `Path` type or a `io.BufferedReader`? I.e.:
```suggestion
    def upload_file(
        self,
        file_inputs: Union[FileSpec, list[FileSpec],
    ) -> Union[dict, requests.Response]:
```
where `FileSpec = Union[str, tuple[str, io.BufferedReader]]` (`tuple` as it seems a path is still required when specifying a file object).

File: meorg_client/tests/test_cli.py [hunk 2]
```diff
@@ -27,23 +27,38 @@ def test_file_upload(runner):
...
+#     """Test file-status via CLI."""
+
+#     # Get the file ID based on the job ID
+#     job_id = store.get("job_id")
+#     result = runner.invoke(cli.file_status, [job_id])
+#     assert result.exit_code == 0
+#     assert result.output != "Pending"
+
+#     # Add file_id to the store for the next test
+#     store.set("file_id", result.output.strip())
```
Comment (SeanBryan51) [inline meorg_client/tests/test_cli.py:63]: Remove unused test case (or skip them with pytest if still relevant?)

```suggestion
```

File: meorg_client/tests/test_client.py [hunk 3]
```diff
@@ -73,7 +79,8 @@ def test_file_list(client):
 
 def test_attach_files_to_model_output(client):
     # Get the file id from the job id
-    file_id = store.get("file_status").get("data").get("files")[0].get("file")
+    # file_id = store.get("file_status").get("data").get("files")[0].get("file")
```
Comment (SeanBryan51) [inline meorg_client/tests/test_client.py:82]: Remove commented line for `file_status`
```suggestion
```

**Review by SeanBryan51 — COMMENTED** [review]
I've added some comments regarding the `upload_files` function. The design of `upload_files` is bugging me a bit. Would it make more sense to have a separate function for uploading in memory file objects? This way I don't think you would need a temporary `_files` variable. Sorry I probably should have raised this in my first review.

**Comment by bschroeter** [comment]
New commit added with the suggested fixes. I think we're in a good spot now, its much cleaner.

**Comment by bschroeter** [comment]
> I've added some comments regarding the `upload_files` function. The design of `upload_files` is bugging me a bit. Would it make more sense to have a separate function for uploading in memory file objects? This way I don't think you would need a temporary `_files` variable. Sorry I probably should have raised this in my first review.

A singular function is implemented. It would be basically a heap of replicated code to handle objects in a separate function. The use case on objects is slim, mostly for testing and direct connection via Python. 99% of the usage will be on the command line with a path that gets opened.

**Review by SeanBryan51 — APPROVED** [review]

**Pull request merged**